### ISSUE TITLE
Appropriate loss function For NxN grid prediction

### ISSUE THREAD

**Issue by jasper95** [fchollet/keras#2065]
I have the ff. model. my input is AxB grid and output flatten AxB

model = Sequential()
model.add(Convolution2D(1, filter_size, filter_size, border_mode='same',
                        input_shape=(1, 107, 72)))
model.add(LeakyReLU())
model.add(Convolution2D(1, filter_size, filter_size, border_mode='same'))
model.add(LeakyReLU())
model.add(Convolution2D(1, filter_size, filter_size, border_mode='same'))
model.add(LeakyReLU())
model.add(Convolution2D(1, filter_size, filter_size, border_mode='same'))
model.add(LeakyReLU())
model.add(Flatten())
model.add(Dense(boxes, activation='softmax'))
model.compile(loss='categorical_crossentropy', optimizer='adam') #i think this is incorrect
checkpoint = ModelCheckpoint('weight.hdf5',
                             verbose=1, save_best_only=True)
model.fit(X_train, Y_train, batch_size=32, nb_epoch=50,
          validation_data=(X_test, Y_test), show_accuracy=True,
          callbacks=[checkpoint])
